### DISCUSSION TITLE
No data returned when rasterizing specific measurements

### DISCUSSION THREAD

**Discussion by lwasser** [corteva/geocube#50] [Q&A] [answered]
hi there @snowman2 and all!
i just discovered this tool. i'm trying to rasterize some trails data to perform a spatial analysis- 

Im encountering a few odd results that are likely user error.

1. i'd  like my raster layer to contains the segment ID's associated with each cell however when i specify measurements, it seems to not return any of that column data.
2. I'd like plot the data but the conventional `.plot()`  fails i presume because all that is in my output raster is coordinates - no data
I'm sure i'm missing a crucual step. I'd like a xarray object with x,y and z (x and y being coordinates and z being a unique trail id from the `GISPROD3OSMPTrailsOSMPSEGMENTID` column. Can you provide any direction on what i'm doing wrong?

my code is below:

```
import geopandas as gpd
from geocube.api.core import make_geocube


all_trails = gpd.read_file("https://opendata.arcgis.com/datasets/d7ad8e150c164c32ab1690658f3fa662_4.geojson")

# Convert data to raster
trails_raster = make_geocube(vector_data=all_trails, 
                    # this measurements parameter doesn't seem to work as i expect it to!?
                    measurements=["GISPROD3OSMPTrailsOSMPSEGMENTID","GISPROD3OSMPTrailsOSMPTRAILNAME"], 
                    resolution=(res, -res))

```

many thanks in advance!!

**Comment by snowman2** [comment, marked as answer]
Welcome!

The main issue is that the columns you specified are not numeric. Non-numeric columns cannot be rasterized and are skipped. If you would like to rasterize them, I would recommend looking at the [categorical data](https://corteva.github.io/geocube/stable/examples/categorical.html) example or to convert the data into a numeric format using a different method that makes sense for your data.

**Comment by lwasser** [comment]
great - thank you!! this absolutely makes sense @snowman2 thank you. and 👋 !! I can convert to numeric no problem in this case.  
I have a followup question - if you would prefer i ask in a new thread, please do say the word. i'm new to discussions in GitHub but am really liking this interface as a way to ask questions.

I have one dataset rasterized using a numeric column - yay!

Now i'm having an issue with a second layer that is one I created by downloading data in JSON format through an api.
The data are lines (trails once again) and in a xy format. a screenshot of the data are below. i could try to recreate this issue however with a smaller dataset if that would make life easier. i'm struggling to do that right now and still recreate the same issue!

<img width="893" alt="Screen Shot 2020-12-25 at 3 06 15 PM" src="https://user-images.githubusercontent.com/7649194/103142598-da991f80-46c2-11eb-8963-e00fb7a6f4f7.png">

When i run make_geocube

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-46-796abdcb7f86> in <module>
      1 # Convert data to raster
----> 2 trails_raster = make_geocube(vector_data=gdf_2020, 
      3                     #measurements=["activity_id"],
      4                     resolution=(res, -res))

~/anaconda3/envs/earth-analytics-python/lib/python3.8/site-packages/geocube/api/core.py in make_geocube(vector_data, measurements, datetime_measurements, output_crs, resolution, align, geom, like, fill, group_by, interpolate_na_method, categorical_enums, rasterize_function)
     84     geobox_maker = GeoBoxMaker(output_crs, resolution, align, geom, like)
     85 
---> 86     return VectorToCube(
     87         vector_data=vector_data,
     88         geobox_maker=geobox_maker,

~/anaconda3/envs/earth-analytics-python/lib/python3.8/site-packages/geocube/vector_to_cube.py in make_geocube(self, measurements, datetime_measurements, group_by, interpolate_na_method, rasterize_function)
    156                 pass
    157 
--> 158         return self._get_dataset(
    159             vector_data, measurements, group_by, interpolate_na_method
    160         )

~/anaconda3/envs/earth-analytics-python/lib/python3.8/site-packages/geocube/vector_to_cube.py in _get_dataset(self, vector_data, measurements, group_by, interpolate_na_method)
    229                 )
    230             else:
--> 231                 grid_array = self._get_grid(
    232                     vector_data[[measurement, "geometry"]], measurement
    233                 )

~/anaconda3/envs/earth-analytics-python/lib/python3.8/site-packages/geocube/vector_to_cube.py in _get_grid(self, dataframe, measurement_name)
    329             self.fill if str(dataframe[measurement_name].dtype) != "category" else -1
    330         )
--> 331         image_data = self.rasterize_function(
    332             geometry_array=dataframe.geometry,
    333             data_values=_format_series_data(dataframe[measurement_name]).values,

~/anaconda3/envs/earth-analytics-python/lib/python3.8/site-packages/geocube/rasterize.py in rasterize_image(geometry_array, data_values, geobox, fill, merge_alg, filter_nan, all_touched, **ignored_kwargs)
     76                 data_values, geometry_array
     77             )
---> 78         image = rasterio.features.rasterize(
     79             zip(geometry_array.apply(mapping).values, data_values),
     80             out_shape=(geobox.height, geobox.width),

~/anaconda3/envs/earth-analytics-python/lib/python3.8/site-packages/rasterio/env.py in wrapper(*args, **kwds)
    383         else:
    384             with Env.from_defaults():
--> 385                 return f(*args, **kwds)
    386     return wrapper
    387 

~/anaconda3/envs/earth-analytics-python/lib/python3.8/site-packages/rasterio/features.py in rasterize(shapes, out_shape, fill, out, transform, all_touched, merge_alg, default_value, dtype)
    318 
    319     if not validate_dtype(shape_values, valid_dtypes):
--> 320         raise ValueError(format_invalid_dtype('shape values'))
    321 
    322     if dtype is None:

ValueError: shape values dtype must be one of: int16, int32, uint8, uint16, uint32, float32, float64
```

This is odd because looking at the data it seems like they are ok - all numeric and likely floating points (lat / lon coordinates) in wgs84 CRS. I tested for empty or NA geometry and that is not the issue. the data also plot just fine. Can you think of anything else I could check to get to the bottom of this error? Many thanks!!

**Reply by snowman2** [reply]
> if you would prefer i ask in a new thread, please do say the word

If you wouldn't mind, that would be helpful for organization. Having one issue per discussion and keeping them on topic will help those looking through the discussions in the future.

**Reply by lwasser** [reply]
if course!  I'll post in a new thread!
